src/ggcomm.c

```diff
@@ -1344,6 +1346,26 @@ concatenate_results (RESULT_NODE *r,
 
     rc [totlines * totlinelength] = 0;
 
+    /*  Trim whitespace from all lines  */
+    txt = rc;
+    while (txt)
+      {
+         line_end = strchr(txt, '\n');
+         if (!line_end)
+           break;
+         if (*(line_end - 1) == '\r')
```

Comment:
Seems like it should cover the two scenarios, but by circumventing the symbol may not cover certain others and is a bit of a hack. But the last time It tried to generalize the line terminator I ended up backing out changes, so if it works I would go with it.